### PR TITLE
[RFC] Faas fields - stage 2 - change faas.trigger type from nested to object

### LINKED ISSUE #2192: Remove mapping for field `faas.trigger`
**Summary**

I would like to remove the mapping for the field `faas.trigger` of type `nested`.

**Motivation**:

- The field faas.trigger can only accept a single object like you can see in the apm code at [here](https://github.com/elastic/apm-agent-java/blob/be3870b03e33e670edaaae35ba0f3584ff3984f2/apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/Faas.java#L41)
- Nested documents and queries are typically expensive
- the field `faas.trigger` is an object in ECS but it maps to a single string in Otel like documented at [issue](https://github.com/elastic/ecs/issues/1516#issuecomment-904365116) and in the Otel [docs](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/faas.md#incoming-invocations)
- Avoiding unnecessary `nested` fields should make it easier to support `subobjects: false` in the project Logs+ for default o11y mappings. 

**Detailed Design**:

Provide additional details around the design of the proposed changes.

* Field names: `faas.trigger`
* Example values for the fields

```json
"trigger": {
          "request_id": "10",
          "type": "http"
      }
  }
```

* Suggested appropriate datatypes

suggested mapping. Notice the change 

```json
{
    "mappings": {
        "properties": {
            "faas": {
                "type": "object"
            },
            "mappings": {
                "properties": {
                    "trigger": {
                        "type": "object"
                    },
                    "mappings": {
                        "properties": {
                            "request_id": {
                                "type": "keyword"
                            },
                            "type": {
                                "type": "keyword"
                            }
                        }
                    }
                }
            }
        }
    }
}
```

* Any example events that map to the proposed use case(s)

```json
{
    "faas": {
        "trigger": {
            "request_id": "10",
            "type": "http"
        }
    }
}
```

or 

```json
{
    "faas.trigger.request_id": "10",
    "faas.trigger.type": "http"
}
```

### PR DESCRIPTION
**Pull request by gsantoro**
- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/ecs/blob/main/CONTRIBUTING.md)?
- For proposing substantial changes or additions to the schema, have you reviewed the [RFC process](https://github.com/elastic/ecs/blob/main/rfcs/README.md)?
- If submitting code/script changes, have you verified all tests pass locally using `make test`?
- If submitting schema/fields updates, have you generated new artifacts by running `make` and committed those changes?
- Is your pull request against main? Unless there is a good reason otherwise, we prefer pull requests against main and will backport as needed.
- Have you added an entry to the [CHANGELOG.next.md](https://github.com/elastic/ecs/blob/main/CHANGELOG.next.md)?

Changes in this PR:
- removed entirely mapping of faas.trigger since it is not needed anymore
- Closes #2192

### CODE REVIEW

File: experimental/generated/beats/fields.ecs.yml [hunk 1]
```diff
@@ -2699,7 +2699,7 @@
       default_field: false
     - name: trigger
       level: extended
-      type: nested
+      type: object
       description: Details about the function trigger.
       default_field: false
```
Comment (felixbarny) [inline experimental/generated/beats/fields.ecs.yml:2704]: Can we just remove the `trigger` field altogether? It's not really necessary as the `trigger.request_id` field already implicitly creates a `trigger` object. Explicitly mapping a field as an `object` is also problematic in combination with `subobjects: false`.

There's already precedent in ECS to not explicitly define object fields. For example:

https://github.com/elastic/ecs/blob/261a873fcb24ceb9e1a37f40d12905549197579c/schemas/log.yml#L67

**Review by felixbarny — APPROVED** [review]

**Review by ebeahan — COMMENTED** [review]
@gsantoro Can you also add an entry into `CHANGELOG.next`?

**Comment by gsantoro** [comment]
hello @ebeahan, do you mind having another look at this PR? thanks

**Review by AlexanderWert — APPROVED** [review]

**Review by ebeahan — APPROVED** [review]
Sorry for the delay - LGTM after the conflicts are sorted.

**Pull request merged**